### PR TITLE
Fix PDF export

### LINKED ISSUE #10102: PDF export XMP annotated PDF gives error message
### JabRef version

5.9 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10 22H2

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. File -> Export all entries
2. XMP annotated PDF
3. When you save the file, the following error message is appearing. No PDF file is exported

See screenshot

afbeelding

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

### PR DESCRIPTION
**Pull request by Luggas4you**
Fixes #10102

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Luggas4you** [comment]
Throws Exception: Method threw 'java.lang.IllegalStateException' exception. Cannot evaluate org.apache.pdfbox.cos.COSDictionary.toString()

File: src/main/java/org/jabref/logic/exporter/XmpPdfExporter.java [hunk 1]
```diff
@@ -1,15 +1,20 @@
 package org.jabref.logic.exporter;
 
-import java.nio.file.Path;
-import java.util.List;
-import java.util.Objects;
+        import java.io.IOException;
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/exporter/XmpPdfExporter.java:3]: intendation off?

File: src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java [hunk 2]
```diff
@@ -143,6 +157,41 @@ void unsuccessfulExportToFileByPath(Path path) throws Exception {
...
+                    contentStream.newLineAtOffset(25, 500);
+                    contentStream.setFont(new PDType1Font(Standard14Fonts.FontName.HELVETICA), 12);
+                    contentStream.showText("This PDF was created by JabRef. It demonstrates the embedding of BibTeX data in PDF files. Please open the file attachments view of your PDF viewer to see the attached files. Note that the normal usage is to embed the BibTeX data in an existing PDF.");
+                    contentStream.endText();
+                }
+                document.save(originalFilePath.toString());
+            }
+            exporter.exportToFileByPath(databaseContext, filePreferences, originalFilePath, abbreviationRepository);
+
+            Files.copy(Paths.get(originalFilePath.toString()), Paths.get(exportedFilePath.toString()), StandardCopyOption.REPLACE_EXISTING);
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java:181]: You already have two Path objects, so you don't need that wrapping here 
```suggestion
            Files.copy(originalFilePath), exportedFilePath), StandardCopyOption.REPLACE_EXISTING);
```

File: src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java [hunk 3]
```diff
@@ -143,6 +157,41 @@ void unsuccessfulExportToFileByPath(Path path) throws Exception {
...
+
+            try (PDDocument importedDocument = Loader.loadPDF(new File(exportedFilePath.toString()))) {
+                assertNotNull(importedDocument, "The imported document should not be null");
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java:184]: I would use JabRef's PDF Importer for this so you can compare the exported entries with the imported entries
PdfEmbeddedBibFileImporterTest

File: src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java (hunk 2 above)
Comment (Luggas4you) [reply src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java]: Fixed

File: src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java (hunk 3 above)
Comment (Luggas4you) [reply src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java]: I added the importer and commit it

**Comment by Luggas4you** [comment]
I am still working on the test, these are the current results if I run the test:

grafik

**Comment by Luggas4you** [comment]
> I am still working on the test, these are the current results if I run the test:
> 
> grafik

I don't know what the "JabRef15179252027836258260pdf" file is.

I need not to use the filePath, but I currently do not know how I can resolve it (I could use one entry but the database contain's multiple)

**Comment by koppor** [comment]
DevCall:

Current functionality: One can write for one PDF the XMP data (and an embedded .bib file) to the PDF:

image

Mass update is possible at

image

---

Currently, File -> Export also offers export to PDF (XMP or embedded .bib file). Which causes an error.

In this PR, a new PDF is created if the PDF is not found.

We looking forward to user feedback.

We should write:

- nested .bib files
- XMP data (maybe only generic - if more than one entry: "JabRef library")
- All entries rendered using the current preview style

We could even embed the PDFs inside the PDFs.

Pro: It is super-easy now. No ZIP-file generation/export and later import.

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**